### PR TITLE
Add pulsar & zk controller . Add "id" param for baseConfig

### PR DESCRIPTION
**Pull request by xiang092689**

### CODE REVIEW

File: paas-dashboard-java/src/main/java/com/paas/dashboard/controller/ZookeeperController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        return storageZooKeeper.getConfig(name);
+    }
+
+
+    @GetMapping("/instances")
+    public Map getConfig() {
+        return storageZooKeeper.getConfigMap();
+    }
+
+}
```
Comment (fu-turer) [inline paas-dashboard-java/src/main/java/com/paas/dashboard/controller/ZookeeperController.java:65]: need a new line at the end of file

File: paas-dashboard-java/src/main/java/com/paas/dashboard/controller/ZookeeperController.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package com.paas.dashboard.controller;
```
Comment (fu-turer) [inline paas-dashboard-java/src/main/java/com/paas/dashboard/controller/ZookeeperController.java:1]: license header

File: paas-dashboard-java/src/main/java/com/paas/dashboard/controller/PulsarController.java [hunk 3]
```diff
@@ -43,4 +48,33 @@ public HttpStatus saveInstance(@RequestBody ReqSaveInstancesVo reqSaveInstancesV
...
         }
     }
+
+    @DeleteMapping("/instance")
+    public HttpStatus deleteInstance(@RequestParam("id") int instanceId) {
+        boolean deleteSuccess = storagePulsar.deleteConfig(instanceId);
+        if (deleteSuccess) {
+            return HttpStatus.OK;
+        } else {
+            return HttpStatus.BAD_REQUEST;
```
Comment (fu-turer) [inline paas-dashboard-java/src/main/java/com/paas/dashboard/controller/PulsarController.java:58]: response code should be exactly

Comment (fu-turer) [inline paas-dashboard-java/src/main/java/com/paas/dashboard/controller/PulsarController.java:58]: all response codes should be exactly

File: paas-dashboard-java/src/main/java/com/paas/dashboard/controller/PulsarController.java [hunk 4]
```diff
@@ -43,4 +48,33 @@ public HttpStatus saveInstance(@RequestBody ReqSaveInstancesVo reqSaveInstancesV
...
+        }
+    }
+
+    @PostMapping("/instance")
+    public HttpStatus updateInstance(@RequestBody ReqSaveInstancesVo target) {
+        boolean updateSuccess = storagePulsar.updateConfig(target);
+        if (updateSuccess) {
+            return HttpStatus.OK;
+        }
+        return HttpStatus.BAD_REQUEST;
```
Comment (fu-turer) [inline paas-dashboard-java/src/main/java/com/paas/dashboard/controller/PulsarController.java:68]: all response codes should be exactly

File: paas-dashboard-java/src/main/java/com/paas/dashboard/storage/AbstractStorage.java [hunk 5]
```diff
@@ -66,6 +67,31 @@ public boolean deleteConfig(String name) {
...
     }
 
+    public boolean deleteConfig(int id) {
+        Map<String, T> configs = this.getConfigMap();
+        boolean isEdited = false;
+        List<String> keys = new ArrayList<>(configs.keySet());
+        for (String key : keys) {
+            T value = configs.get(key);
+            if (value.getId() == id) {
+                configs.remove(key);
```
Comment (fu-turer) [inline paas-dashboard-java/src/main/java/com/paas/dashboard/storage/AbstractStorage.java:77]: it may be throw `ConcurrentModificationException`

Comment (xiang092689) [reply paas-dashboard-java/src/main/java/com/paas/dashboard/storage/AbstractStorage.java:77]: it is already copied in a list

**Pull request closed without merging**